Utils/helpers.js

```diff
@@ -39,4 +43,20 @@ const createConversation = async function (currentUserId, searchedUserId) {
   }
 }
 
-module.exports = { checkPassword, createConversation }
+const decodeJWT = async function (token, jwtENV) {
+  const decoded = await promisify(jwt.verify)(token, jwtENV)
+
+  const currentUser = await User.findOne({
+    raw: true,
+    where: {
+      id: decoded.id
+    }
+  })
+  if (!currentUser) {
+    return new AppError('The user does not exist', 401)
+  }
+
+  return currentUser.username
+}
+
+module.exports = { checkPassword, createConversation, decodeJWT }
```

Comment:
DRY. If have the same exact code in the AuthController protect method(btw you should move that out of the controller and into a auth middleware file). Put this into a helper function.